OnlineStore/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using BLL.DTO.Request.Category;
+using BLL.DTO.Response;
+using BLL.Service.Interface;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OnlineStore.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CategoryController : ControllerBase
+    {
+        private readonly ICategoryService _categoryService;
+
+        public CategoryController(ICategoryService categoryService)
+        {
+            _categoryService = categoryService;
+        }
+
+        // GET: api/category
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<CategoryDto>>> GetCategories(CancellationToken cancellationToken)
+        {
+            var categories = await _categoryService.GetAllCategoriesAsync(cancellationToken);
+            return Ok(categories);
+        }
+
+        // GET: api/category/{id}
+        [HttpGet("{id}")]
+        public async Task<ActionResult<CategoryDto>> GetCategory(Guid id, CancellationToken cancellationToken)
+        {
+            try
```

Comment:
remove the try catch in the controllers. There is a global exception handler for this, controllers must be small